src/components/chatbox/chatbox.tsx

```diff
@@ -0,0 +1,550 @@
+		<div className={cn(s.chat)}>
+			{messageList.length === 0 ? (
+				<WelcomeMessage />
+			) : (
+				<div
+					ref={mergeRefs([textContentRef, textContentRef2])}
+					className={s.message_list}
+				>
+					{messageList.map((e, i) => {
+						switch (e.type) {
+							// User input
+							case 'user': {
+								return (
+									<UserMessage
+										key={`${e.type}-${i}`}
+										image={e.image}
+										text={e.text}
+									/>
+								)
+							}
+							// Backend AI response
+							case 'assistant': {
+								const shouldShowActions = e.text?.length > 20 && !isLoading
+								return (
+									<AssistantMessage
+										markdown={e.text}
+										key={e.messageId}
+										conversationId={e.conversationId}
+										messageId={e.messageId}
+										showActions={shouldShowActions}
+									/>
+								)
+							}
+							// Application message; likely an error
+							case 'application': {
+								return (
+									<ApplicationMessage key={`${e.type}-${i}`} text={e.text} />
+								)
+							}
+						}
+					})}
+				</div>
+			)}
+
+			<form onSubmit={handleSubmit} ref={formRef}>
+				<div className={s.bottom}>
+					{textContentScrollBarIsVisible ? (
+						<div className={s.arrowdown} onClick={handleArrowDownClick}>
+							<IconTile size="small">
+								<IconArrowDownCircle16 />
+							</IconTile>
+						</div>
+					) : null}
+
+					<div className={s.question}>
+						<div className={s.question_icon}>
+							{isLoading ? (
+								<IconLoading24 className={'loading'} />
+							) : (
+								<IconWand24
+									style={{
+										color: 'var(--token-color-foreground-highlight-on-surface)',
+									}}
+								/>
+							)}
+						</div>
+						<textarea
+							autoFocus
+							className={cn(s.reset, s.textarea)}
+							spellCheck={false}
+							value={userInput}
+							onChange={(e) => setUserInput(e.currentTarget.value)}
+							onKeyDown={(e) => {
+								// enter submits form; // shift+enter adds a newline
+								if (e.key == 'Enter' && e.shiftKey == false) {
+									if (userInput) {
+										e.preventDefault()
+										formRef.current.requestSubmit()
+									}
+								}
+							}}
+							id="task"
+							rows={userInput.split('\n').length}
+							placeholder="Send a new message"
+							disabled={isLoading}
+						/>
+						{isLoading ? (
+							<Button
+								type={'button'}
+								icon={<IconStopCircle24 height={16} width={16} />}
+								text={'Stop generating'}
+								color={'secondary'}
+								onClick={stopStream}
+							/>
+						) : (
+							<Button
+								disabled={userInput.length < 1}
+								type={'submit'}
+								icon={<IconSend24 height={16} width={16} />}
+								text={'Send'}
+							/>
+						)}
+					</div>
+
+					<Text
+						asElement="span"
+						className={s.disclaimer}
+						/* Display/100/Regular */
+						size={100}
+						weight="regular"
+					>
+						AI Disclaimer: HashiCorp AI may produce inaccurate information and
```

Comment:
I personally love this, but do we need to adjust the language before merging this?

Reply:
:trollface: Unfortunately, we _do_ have to adjust this 

---

> [!IMPORTANT]
> @thiskevinwang work with @kellyschummer on this copy